src/connection.js

```diff
@@ -39,6 +39,12 @@ class Connection extends EventEmitter {
     super();
 
     this.config = config;
+
+    if (typeof (config.domain) === 'string') {
+      this.config = JSON.parse(JSON.stringify(config));
```

Comment:
I don't think this line here is quite right - what's the reason for serializing and deserializing the config here?